zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -578,10 +591,6 @@ private void errorHandlingInTransaction(final Throwable processingException) thr
                   processingException, typedCommand, processingResultBuilder);
           pendingWrites = currentProcessingResult.getRecordBatch().entries();
           pendingResponses = currentProcessingResult.getProcessingResponse().stream().toList();
-          // we need to mark the command as processed, even if the processing failed
-          // otherwise we might replay the events, which have been written during
-          // #onProcessingError again on restart
-          finalizeCommandProcessing();
```

Comment:
:thought_balloon: This is the only change I'm unsure about. `errorHandlingInTransaction` is called when `updateState` fails to commit the transaction. Afterwards, we just call `updateState` again. I'm afraid there is a preexisting bug here because `errorHandlingInTransaction` produces a new `currentProcessingResult`. But in this code path, nothing ever tries to write this: we don't call `writeRecords`. Because we moved the `finalizeCommandProcessing` call to `writeRecords` we have no made this preexisting bug worse, not only do we not write the results from handling the error, we also don't mark the command as processed.

Reply:
Actually it doesn't matter for this change because `updateState` is called after we've written and thus called `finalizeCommandProcessing` already. So yes, there might be an old bug where `updateState` is not writing new processing results from error handling in case the transaction can't be committed but that's a different problem.

Reply:
I've discussed this with @Zelldon and we both came to the conclusion that this is an old bug (that might never be triggered though). I reported it here: https://github.com/camunda/camunda/issues/24675

Reply:
OK. Leaving this as is then.